sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -236,6 +236,11 @@ case class CreateDataSourceTableAsSelectCommand(
               existingSchema = Some(l.schema)
             case s: SimpleCatalogRelation if DDLUtils.isDatasourceTable(s.metadata) =>
               existingSchema = Some(DDLUtils.getSchemaFromTableProperties(s.metadata))
+            // When the source table is a Hive table, the `saveAsTable` API of DataFrameWriter
+            // does not work. Instead, use the `insertInto` API.
+            case o if o.getClass.getName == "org.apache.spark.sql.hive.MetastoreRelation" =>
```

Comment:
it's really hacky to compare with the class name string...  What's the error message before?